appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js

```diff
@@ -239,20 +246,104 @@ export async function postIPComment(request, response) {
  * @param {import('@pins/express/types/express.js').Request} request
  * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
  */
-export async function renderCheckYourAnswers(request, response) {
-	const pageContent = checkYourAnswersPage(
-		request.currentAppeal,
-		request.session?.addIpComment,
-		request.session?.fileUploadInfo,
-		request.errors
-	);
+export async function renderCheckYourAnswers(
+	{
+		errors,
+		currentAppeal: { appealReference, appealId } = {},
+		currentComment: { id: commentId } = {},
+		session: {
+			fileUploadInfo: { files: [{ name, blobStoreUrl }] } = {
+				files: [{ name: '', blobStoreUrl: '' }]
+			},
+			addIpComment: {
+				[redactionStatusFieldName]: redactionStatus,
+				day,
+				month,
+				year,
+				firstName,
+				lastName,
+				emailAddress,
+				addressProvided,
+				addressLine1,
+				addressLine2,
+				town,
+				county,
+				postCode
+			} = {
+				redactionStatus: 'unredacted',
+				day: '',
+				month: '',
+				year: '',
+				firstName: '',
+				lastName: '',
+				emailAddress: '',
+				addressProvided: '',
+				addressLine1: '',
+				addressLine2: '',
+				town: '',
+				county: '',
+				postCode: ''
+			}
+		}
+	},
+	response
+) {
+	if (!isValidRedactionStatus(redactionStatus)) {
+		throw new Error('Received invalid redaction status');
+	}
 
-	return response
-		.status(request.errors ? 400 : 200)
-		.render('patterns/check-and-confirm-page.pattern.njk', {
-			errors: request.errors,
-			pageContent
-		});
+	return renderCheckYourAnswersComponent(
```

Comment:
The problem with moving this out of the mapper into the controller is you're losing the separation of concerns and encapsulation of logic. This function is so generic you basically have to pass the entire contents of the old mapper function as parameters, at which point you may as well not have the helper and just define the CYA page components inline in the controller code.